pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -185,6 +185,7 @@ type attachedVolume struct {
 	nodesAttachedTo map[types.NodeName]nodeAttachedTo
 
 	// devicePath contains the path on the node where the volume is attached
+	// TODO-P0: What if we are attached to multiple nodes?
```

Comment:
Ah, you remind me of this issue we left out earlier. I think this is an issue for especially AWS because device path is dynamically assigned. For GCE PD, it is ok because device path is static and we construct it from the node side. For AWS, if the device path from different node attachment is different, the device path from the last attachement will be used. This problem will require some data structure changes including Node.Status.VolumesAttached. Currently it only has one device path for each volume.